### PR TITLE
add typing-extensions

### PR DESCRIPTION
**Pull request by dougalsutherland**
Needed for the new version of `onnx`.

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
Thanks! amazed this isn't here yet.

File: recipes/typing-extensions/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - typing >=3.6.2  # [py<3.5]
```
Comment (mariusvniekerk) [inline recipes/typing-extensions/meta.yaml]: Pity about this selector making it not noarch

Comment (dougalsutherland) [reply recipes/typing-extensions/meta.yaml]: The source [is actually different](https://github.com/python/typing/blob/ff2084eb9ab5eea778d744c706e6a6aad05ea4bd/typing_extensions/setup.py#L47) for python 2 and 3 anyway.

**Comment by dougalsutherland** [comment]
Blegh, pypi prefers the name `typing-extensions` but has files under `typing_extensions`.

I'll make a subpackage for the dashed name....

**Comment by dougalsutherland** [comment]
Linter error is a false positive; would be fixed by https://github.com/conda-forge/conda-smithy/pull/830.

**Comment by dougalsutherland** [comment]
No, I didn't mean that the package is `import typing_extensions`; I meant that the file source is `typing_extensions-3.6.5.tar.gz`.

**Comment by dougalsutherland** [comment]
@conda-forge/staged-recipes, opinions about the metapackage? I can remove and only have a dashed version if people prefer that, but given that both versions of the name are used for `pip` and there's essentially no overhead to it, I think it makes some sense to have both.

**Comment by ocefpaf** [comment]
> I think it makes some sense to have both.

I'm OK with that.

**Pull request merged**